frontend.sh

```diff
@@ -15,25 +15,34 @@ init() {
   source cli-flag.sh 'Front end' $1
 
   NODE_DIR=node_modules
+  DEP_CHECKSUM=`cat npm-shrinkwrap.json package.json | shasum -a 256`
 
   echo "npm components directory: $NODE_DIR"
 }
 
-# Clean project dependencies.
+# Clean project dependencies if needed.
 clean() {
-  # If the node directory already exists,
-  # clear it so we know we're working with a clean
-  # slate of the dependencies listed in package.json.
+  # If the node directory exists, $NODE_DIR/CHECKSUM exists, and
+  # the contents DO NOT match the checksum of package.json, clear
+  # $NODE_DIR so we know we're working with a clean slate of the
+  # dependencies listed in package.json.
   if [ -d $NODE_DIR ]; then
-    echo 'Removing project dependency directories...'
-    rm -rf $NODE_DIR
+    echo $DEP_CHECKSUM
+    echo `cat $NODE_DIR/CHECKSUM`
+    if [ ! -f $NODE_DIR/CHECKSUM ] || 
+       [ "$DEP_CHECKSUM" != "`cat $NODE_DIR/CHECKSUM`" ]; then
+      echo 'Removing project dependency directories…'
+      rm -rf $NODE_DIR
+      echo 'Project dependencies have been removed.'
+    else
+      echo 'Project dependencies checksum matches package.json…'
+    fi
   fi
-  echo 'Project dependencies have been removed.'
 }
 
 # Install project dependencies.
 install() {
```

Comment:
@sebworks if we're not deleting the node_modules directory and package.json and npm-shrinkwrap.json haven't changed is there a need to run `install` here at all? That could also reduce the amount of time spent.

So then we check for the existence and validity of the checksum, and if it passes we just build.